server.py

```diff
@@ -320,6 +367,7 @@ def queue_quit_message_for_other_users(self, quit_message: str) -> None:
             receiver.send_que.put((quit_message, self.get_user_mask()))
 
     def get_users_sharing_channel(self) -> Set[UserConnection]:
+        """Returns a set of all users sharing a channel with the client."""
```

Comment:
You are repeating what the method name already says. As it is currently, the docstring doesn't provide any useful information to the reader. You can delete the docstring, or reword it:

```suggestion
        """Returns all users of all channels that this user has joined."""
```